src/compiler/checker.ts

```diff
@@ -2401,6 +2463,22 @@ namespace ts {
                         return getReturnTypeOfSignature(getSignatureFromDeclaration(getter));
                     }
                 }
+
+                // In a JS file, we might need to add types to parameters of a function expression that is
+                // an argument to 'define'
+                if (func.kind === SyntaxKind.FunctionExpression &&
+                    func.parent &&
+                    func.parent.kind === SyntaxKind.CallExpression) {
+                    
+                    if (isDefineCall(<CallExpression>func.parent) || isAmdRequireCall(<CallExpression>func.parent)) {
+                        assignDefineOrRequireCallParameterTypes(<FunctionExpression>func);

```

Comment:
:heavy_check_mark: I think. Completely removed `assignDefineOrRequireCallParameterTypes` and moved logic to parts of the contextual typing system